src/Microsoft.AspNet.Mvc.Razor/RazorView.cs

```diff
@@ -233,13 +238,18 @@ private Task RenderPageCoreAsync(IRazorPage page, ViewContext context)
             }
         }
 
-        private IRazorPage GetLayoutPage(ViewContext context, string layoutPath)
+        private IRazorPage GetLayoutPage(ViewContext context, string executingFilePath, string layoutPath)
         {
-            var layoutPageResult = _viewEngine.FindPage(context, layoutPath);
+            var layoutPageResult = _viewEngine.GetPage(executingFilePath, layoutPath, isPartial: true);
```

Comment:
@pranavkm the previous version didn't _expose_ the `IsPartial` notion. Under the covers however, it returned a page with `IsPartial` set to `true`. What I've done here is make the notion visible and make the `[Find|Get]Page` methods directly usable from `[Find|Get]View`. That's already been useful because it spurred the conversation @Eilon and I have been having about whether `true` is correct for layouts.